### ISSUE TITLE
Entry TTL is not updated if EntryProcessor implements Offloadable

### ISSUE THREAD

**Issue by eleftherias** [hazelcast/hazelcast#17606]
**Describe the bug**
Hazelcast 4.0.3 added the ability to set the expiration times for entries when using entry processors, with the addition of `ExtendedMapEntry`.
The suggested configuration does not update the TTL if the `EntryProcessor` implements the `Offloadable` interface.

**Expected behavior**
The entry expiration time should be updated to the new TTL.

**To Reproduce**

```java
private static class TTLChangingEntryProcessor<K, V> implements EntryProcessor<K, V, V>, Offloadable {

    private V newValue;
    private Duration newTtl;

    TTLChangingEntryProcessor(V newValue, Duration newTtl) {
        this.newValue = newValue;
        this.newTtl = newTtl;
    }

    @Override
    public V process(Entry<K, V> entry) {
        return ((ExtendedMapEntry<K, V>) entry).setValue(newValue, newTtl.toMillis(), TimeUnit.MILLISECONDS);
    }

    @Override
    public String getExecutorName() {
        return OFFLOADABLE_EXECUTOR;
    }

}
```

**Additional context**

Related: #16987

**Comment by ahmetmircik** [comment]
This is not appropriate to solve this for Offloadables in 4.0.x series due to the backward compatibility reasons. But in 4.1, we can fix this.

2 things we can do here:
 - To add documentation for this exception in 4.0.x series.
 - To fix the issue in next 4.1 release.

**Comment by Holmistr** [comment]
Hi @eleftherias , thanks a lot for the report. We'll take a look. We're heading towards the 4.1 release in quite a tight schedule, so can't make any promises if it gets in :) But we'll keep you posted here.
